ui/src/components/modal/dynamic-modal.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
+
+import {
+    AlertDialog,
+    AlertDialogHeader,
+    AlertDialogContent,
+    AlertDialogDescription,
+    AlertDialogTitle,
+    AlertDialogFooter,
+    AlertDialogCancel
+} from '@/components/ui/alert-dialog';
+import { ReactNode, useState } from 'react';
+import { Button } from '@/components/ui/button';
+
+const DynamicModal = ({
+    open,
+    title,
+    body,
+    buttons
+}: {
+    open: boolean;
+    title: string;
+    body: string;
+    buttons: ReactNode[];
```

Comment:
Maybe we should limit the buttons in the dynamicModal to only two at max? Therefore we could call this the secondaryButton button and let it be optional.

Reply:
If we limit the number of buttons to two, is it guaranteed that we won't use a third in the future? I thought it'd be best to leave the number of buttons open to future usages since I noticed that in the AngularJS implementation, there is only the option to close it. Therefore, we aren't able to use it for the ApiErrorModal since it needs another button for the Feedback page which may defeat the purpose of a dynamic modal. 

I was also thinking about the Timeout modal and how the dynamic modal can be used for that purpose. I think we can keep the Timeout modal file because it has external logic that should be separated from its parent component (the navbar) but call the dynamic modal in the return statement. I can look into that more because I know that the timer needs to count down and not sure if it'll uphold that.

Reply:
I've never seen a modal with more than two buttons.

Reply:
Same here 😆 
However, if you create two buttons only, you need to check whether or not both have been passed in and you're adding an additional parameter. Whereas right now, the mapping function will only render a button if it exists naturally.

Reply:
Just make sure to set buttons as optional: `buttons?: ReactNode[]`, thanks.